frontend/lib/types/BasicClassType.cpp

```diff
@@ -62,6 +62,19 @@ BasicClassType::getObjectType(Context* context) {
                            SubstitutionsMap()).get();
 }
 
+const BasicClassType*
+BasicClassType::getReduceScanOpType(Context* context) {
+  auto symbolPath = UniqueString::get(context, "ChapelReduce.ReduceScanOp");
+  auto name = UniqueString::get(context, "ReduceScanOp");
+  auto id = ID(symbolPath, -1, 0);
+  auto objectType = getObjectType(context);
+
+  return getBasicClassType(context, id, name,
+                           /* parentType */ objectType,
+                           /* instantiatedFrom */ nullptr,
+                           SubstitutionsMap()).get();
+}
```

Comment:
FYI, I think @arezaii's PR will move some similar functionality to CompositeType. We need to track these types also in a helper function (currently `isMissingBundledRecordType`) to get compilation working without the standard library.

Reply:
I'm coordinating with Ahmad to get the code written and refactored smoothly.